servicetalk-http-api/src/main/java/io/servicetalk/http/api/BlockingStreamingToStreamingService.java

```diff
@@ -197,11 +202,61 @@ public void close(final Throwable cause) throws IOException {
 
         @Override
         public HttpHeaders trailers() {
+            if (trailers == null) {
+                trailers = trailersFactory.get();
+            }
+            return trailers;
+        }
+
+        @Nullable
+        HttpHeaders trailers0() {
             return trailers;
         }
 
         Publisher<Buffer> connect() {
             return payloadWriter.connect();
         }
     }
+
+    private static final class TrailersMapper implements ScanMapper<Object, Object>, ScanMapper.MappedTerminal<Object> {
+        private final BufferHttpPayloadWriter payloadWriter;
+
+        TrailersMapper(final BufferHttpPayloadWriter payloadWriter) {
+            this.payloadWriter = payloadWriter;
+        }
+
+        @Nullable
+        @Override
+        public Object mapOnNext(@Nullable final Object next) {
+            return next;
+        }
+
+        @Nullable
+        @Override
+        public MappedTerminal<Object> mapOnError(final Throwable cause) {
+            return null;
+        }
+
+        @Override
+        public MappedTerminal<Object> mapOnComplete() {
+            return this;
+        }
+
+        @Nullable
+        @Override
+        public Object onNext() {
+            return payloadWriter.trailers0();
+        }
+
+        @Override
+        public boolean onNextValid() {
+            return payloadWriter.trailers0() != null;
```

Comment:
see below question on the use-case for emptry trailers. If not, we could also check here that it's not null and also not empty?